draft-ietf-quic-transport.md

```diff
@@ -1131,7 +1197,34 @@ version negotiation occurred but it would have selected a different version
 based on the value of the supported_versions list.
 
 
-## Proof of Source Address Ownership {#source-address-token}
+## Stateless Retries {#stateless-retry}
+
+A server can process a the initial cryptographic handshake messages from a
+client without committing any state.  This allows a server to perform address
+validation ({{address-validation}}, or to defer connection establishment costs.
+
+In order to ensure consistent state regarding stream offsets and packet numbers,
+a server that generates a response to an initial packet without retaining
+connection state MUST use the Server Stateless Retry packet
+({{packet-server-stateless}}).  This packet causes a client to reset the state
+of stream 0 and continue the connection attempt with fresh stream state while
+maintaining the state of the cryptographic handshake.
+
+A server that sends a Server Stateless Retry packet is then not obligated to
+ensure that it correctly accounts for data already sent on stream 0 either by
+both itself or by the client.  Similarly, a server also does not have to
+remember a randomized initial packet number: the Server Stateless Retry packet
+includes a copy of the packet number from the client packet.
```

Comment:
This is still confusing to me. It's not clear to me why there's even a discussion of what the server does not have to remember. Stateless Retry is a shoot and forget packet, that should be described as "no state is maintained for the received client packet," which should be adequate, I think. Here's what I'm imagining for this section:

## Stateless Retries {#stateless-retry}

A server can process an initial cryptographic handshake messages from a client without committing any state.  This allows a server to perform address validation ({{address-validation}}, or to defer connection establishment costs.

A server that generates a response to an initial packet without retaining connection state MUST use the Server Stateless Retry packet ({{packet-server-stateless}}).  This packet causes a client to reset its transport state and to continue the connection attempt with new connection state while maintaining the state of the cryptographic handshake.

A server MUST NOT send multiple Server Stateless Retry packets in response to a client handshake packet.  Thus, any cryptographic handshake message that is sent MUST fit within a single packet.

In TLS, the Server Stateless Retry packet type is used to carry the HelloRetryRequest message.